decoder/expr_map_completion_test.go

```diff
@@ -0,0 +1,996 @@
+package decoder
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/hcl-lang/lang"
+	"github.com/hashicorp/hcl-lang/schema"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+)
+
+func TestCompletionAtPos_exprTypeDeclaration(t *testing.T) {
+	testCases := []struct {
+		testName           string
+		attrSchema         map[string]*schema.AttributeSchema
+		cfg                string
+		pos                hcl.Pos
+		expectedCandidates lang.Candidates
+	}{
+		{
```

Comment:
We could add one more test case with an empty configuration, like

```
`
`,
```
to test the full attribute completion `attr = {\n ${1}\n}`

Reply:
We can, but that is then testing a different code path, so not the same "unit" as all the other tests.

Reply:
Oh, right. Please ignore my suggestion